executor/show.go

```diff
@@ -1414,35 +1443,39 @@ func ConstructResultOfShowCreateDatabase(ctx sessionctx.Context, dbInfo *model.D
 	sqlMode := ctx.GetSessionVars().SQLMode
 	var ifNotExistsStr string
 	if ifNotExists {
-		ifNotExistsStr = "/*!32312 IF NOT EXISTS*/ "
+		if !ctx.GetSessionVars().EnableSimplifiedShowCreateTable {
+			ifNotExistsStr = "/*!32312 IF NOT EXISTS*/ "
+		} else {
+			ifNotExistsStr = "IF NOT EXISTS "
+		}
```

Comment:
This can be version-comment-wrapped too?
```suggestion
wrapInVersionComment(ctx, buf, "32312", "IF NOT EXISTS")
```

Reply:
Hmm, there are two minor issues, with the comment, it will also end without a space between `EXISTS` and `*/` and it is used as a string in `fmt.Fprintf(buf, "CREATE DATABASE %s%s", ifNotExistsStr,...`
But instead of trying to fix that, how about just remove the comment, since I doubt there will ever be a case where compatibility between TiDB and MySQL < 3.23.12 will be required :)

@morgo, do you agree to remove the version comment instead?

Reply:
I think it's fine. It might break some MySQL tests, but they can be re-recorded.

Reply:
Also looks like this is the only place where we had this v3.2x.x stuff:
```
$ git grep -E '!32[0-9]{3}' | grep -v '_test.go'
executor/show.go:		ifNotExistsStr = "/*!32312 IF NOT EXISTS*/ "
```

There is also some v4.x stuff:
```
$ git grep -E '!4[0-9]{4}' | grep -v '_test.go' | grep -v '/tests/' | grep -v mocker
dumpling/export/dump.go:			"/*!40101 SET NAMES binary*/;",
dumpling/export/ir.go:		specCmts:      []string{"/*!40101 SET NAMES binary*/;"},
dumpling/export/sql.go:	query = "START TRANSACTION /*!40108 WITH CONSISTENT SNAPSHOT */"
dumpling/export/writer.go:			"/*!40101 SET NAMES binary*/;",
executor/show.go:		fmt.Fprintf(buf, " /*!40100 DEFAULT CHARACTER SET %s ", dbInfo.Charset)
executor/show.go:		fmt.Fprintf(buf, " /*!40100 DEFAULT CHARACTER SET %s ", collInfo.CharsetName)
util/dbutil/common.go:	query := fmt.Sprintf("SELECT /*!40001 SQL_NO_CACHE */ MIN(%s%s) as MIN, MAX(%s%s) as MAX FROM %s WHERE %s",
```
It may make sense to clean that up as well.

Reply:
@dveeden I created https://github.com/pingcap/tidb/issues/36270 to track this, to avoid increasing the scope of this PR.